### PR TITLE
StormCast training code improvements

### PR DESCRIPTION
**Pull request by pzharrington**
# Modulus Pull Request

## Description
Refactors the StormCast training recipe to be more in-line with Modulus examples and conventions. Notable changes:
 - Use a hydra-based config management system instead of `YParams`
 - Use Modulus neural networks (`SongUNet`, `EDMPrecond`) and add a regression U-Net wrapper for StormCast (`StormCastUNet`)
 - Use Modulus EDM loss and sampling routines
 - Use Modulus utilities for checkpointing and logging

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is
up to date with these changes.

### CODE REVIEW

File: modulus/models/diffusion/preconditioning.py [hunk 1]
```diff
@@ -584,11 +594,20 @@ def __init__(
         sigma_max=float("inf"),
         sigma_data=0.5,
         model_type="DhariwalUNet",
+        img_in_channels=None,
+        img_out_channels=None,
```
Comment (mnabian) [inline modulus/models/diffusion/preconditioning.py:598]: I'm a little bit worried that this change might break some of the in-progress PRs for contributing other diffusion models to Modulus. I'll have to check

File: examples/generative/stormcast/README.md
Comment (mnabian) [inline examples/generative/stormcast/README.md:1]: Not for this PR but similar to CorrDiff-mini, it would be really useful if we put together a stormcast-mini recipe so that users can quickly test the model. We can also upload a small sample dataset on NGC for users to try out.

**Comment by mnabian** [comment]
Hey @pzharrington , I mostly looked at the changes to the sampler, preconditioner UNet, and metrics to make sure changes are backward compatible and are not breaking other codes. I also briefly looked at the isolated changes for the stormcast recipe and everything look good to me. Approving

**Review by mnabian — APPROVED** [review]
